javers-core/src/main/java/org/javers/core/metamodel/type/TypeMapperState.java

```diff
@@ -155,15 +161,20 @@ private void addFullMapping(Type javaType, JaversType newType){
     /**
      * must be called within synchronized block
      */
-    private JaversType infer(Type javaType) {
+    private JaversType infer(Type javaType, boolean asShallowReference) {
```

Comment:
rzeczywiście z ShallowCdoWrapperem wygląda to dużo lepiej. Skoro przejąłeś implementację w nowym branchu, to ten zamykam.